PWGHF/Utils/utilsMcMatching.h

```diff
@@ -0,0 +1,269 @@
+  {DecayChannelResonant::DstarToD0ToRhoplusK, {+kRho770Plus, +kKMinus}},
+  {DecayChannelResonant::DstarToD0ToKstar0Pi0, {-o2::constants::physics::kK0Star892, +kPi0}},
+  {DecayChannelResonant::DstarToD0ToKstarPi, {-o2::constants::physics::kKPlusStar892, +kPiPlus}},
+  {DecayChannelResonant::DstarToDplusToPhiPi, {+o2::constants::physics::kPhi, +kPiPlus}},
+  {DecayChannelResonant::DstarToDplusToKstar0K, {-o2::constants::physics::kK0Star892, +kKPlus}},
+  {DecayChannelResonant::DstarToDplusToKstar1430_0K, {+10311, +kKPlus}},
+  {DecayChannelResonant::DstarToDplusToRho0Pi, {+kRho770_0, +kPiPlus}},
+  {DecayChannelResonant::DstarToDplusToF2_1270Pi, {+225, +kPiPlus}},
+};
+
+// Lc
+static const std::unordered_map<DecayChannelMain, const std::vector<int>> daughtersLcMain{
+  {DecayChannelMain::LcToPKPi, {+kProton, +kKMinus, +kPiPlus}},
+  {DecayChannelMain::LcToPKPiPi0, {+kProton, +kKMinus, +kPiPlus, +kPi0}},
+  {DecayChannelMain::LcToPPiPi, {+kProton, +kPiMinus, +kPiPlus}},
+  {DecayChannelMain::LcToPKK, {+kProton, +kKMinus, +kKPlus}}};
+
+static const std::unordered_map<DecayChannelResonant, const std::array<int, 2>> daughtersLcResonant{
+  {DecayChannelResonant::LcToPKstar0, {+o2::constants::physics::kK0Star892, +kProton}},
+  {DecayChannelResonant::LcToDeltaplusplusK, {+2224, +kKMinus}},
+  {DecayChannelResonant::LcToL1520Pi, {+102134, +kPiPlus}},
+};
+
+// Xic
+static const std::unordered_map<DecayChannelMain, const std::vector<int>> daughtersXicMain{
+  {DecayChannelMain::XicToPKPi, {+kProton, +kKMinus, +kPiPlus}},
+  {DecayChannelMain::XicToPKK, {+kProton, +kKMinus, +kKPlus}},
+  {DecayChannelMain::XicToSPiPi, {+kSigmaPlus, +kPiMinus, +kPiPlus}},
+};
+
+static const std::unordered_map<DecayChannelResonant, const std::array<int, 2>> daughtersXicResonant{
+  {DecayChannelResonant::XicToPKstar0, {-o2::constants::physics::kK0Star892, +kProton}},
+  {DecayChannelResonant::XicToPPhi, {+kProton, +o2::constants::physics::kPhi}},
+};
+
+/// Returns a map of the possible final states for a specific 3-prong particle specie
+/// \param pdgMother PDG code of the mother particle
+/// \return a map of final states with their corresponding PDG codes
+inline std::unordered_map<DecayChannelMain, const std::vector<int>> getDecayChannelMain(int pdgMother)
+{
+  switch (pdgMother) {
+    case o2::constants::physics::Pdg::kDPlus:
+      return daughtersDplusMain;
+    case o2::constants::physics::Pdg::kDS:
+      return daughtersDsMain;
+    case o2::constants::physics::Pdg::kDStar:
+      return daughtersDstarMain;
+    case o2::constants::physics::Pdg::kLambdaCPlus:
+      return daughtersLcMain;
+    case o2::constants::physics::Pdg::kXiCPlus:
+      return daughtersXicMain;
+    default:
+      LOG(fatal) << "Unknown PDG code for 3-prong final states: " << pdgMother;
+      return {};
+  }
+}
+} // namespace hf_cand_3prong
+
+/// Perform the matching for a single resonant channel
+/// \tparam N size of the array of daughter PDG codes
```

Comment:
The function has nothing to do with resonant channels per se, as I pointed out earlier.